example/lib/src/file_picker_demo.dart

```diff
@@ -50,17 +93,78 @@ class _FilePickerDemoState extends State<FilePickerDemo> {
         withData: true,
       ))
           ?.files;
+      hasUserAborted = pickedFiles == null;
+    } on PlatformException catch (e) {
+      _logException('Unsupported operation' + e.toString());
+    } catch (e) {
+      _logException(e.toString());
+    }
+    if (!mounted) return;
+
+    setState(() {
+      _isLoading = false;
+      _userAborted = hasUserAborted;
+      _resultsWidget = _buildFilePickerResultsWidget(
+        itemCount: pickedFiles?.length ?? 0,
+        itemBuilder: (BuildContext context, int index) {
+          final path =
+              pickedFiles!.map((e) => e.path).toList()[index].toString();
+          return ListTile(
+            leading: Text(
+              index.toString(),
+              style: TextStyle(fontSize: 15, fontWeight: FontWeight.bold),
+            ),
+            title: Text("File path:"),
+            subtitle: Text(path),
+          );
+        },
+      );
+    });
+  }
+
+  void _pickFileAndDirectoryPaths() async {
+    List<String>? pickedFilesAndDirectories;
+    bool hasUserAborted = true;
+    _resetState();
+
+    try {
+      pickedFilesAndDirectories =
+          await FilePicker.platform.pickFileAndDirectoryPaths(
+        type: _pickingType,
+        allowedExtensions: (_extension?.isNotEmpty ?? false)
+            ? _extension?.replaceAll(' ', '').split(',')
+            : null,
+        initialDirectory: _initialDirectoryController.text,
+      );
+      hasUserAborted = pickedFilesAndDirectories == null;
     } on PlatformException catch (e) {
       _logException('Unsupported operation' + e.toString());
     } catch (e) {
       _logException(e.toString());
     }
     if (!mounted) return;
+
     setState(() {
       _isLoading = false;
-      _fileName =
-          _paths != null ? _paths!.map((e) => e.name).toString() : '...';
-      _userAborted = _paths == null;
+      _userAborted = hasUserAborted;
+      _resultsWidget = _buildFilePickerResultsWidget(
+        itemCount: pickedFilesAndDirectories?.length ?? 0,
+        itemBuilder: (BuildContext context, int index) {
+          final String name =
+              new io.File(pickedFilesAndDirectories![index]).statSync().type ==
```

Comment:
Avoid using `new` as it is redundant since Dart 2.0

```suggestion
              new io.File(pickedFilesAndDirectories![index]).statSync().type ==
```